swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/model/Resource.kt

```diff
@@ -171,6 +173,50 @@ interface Identifiable : Named {
   fun resourceUrl(workConfiguration: WorkConfiguration): String {
     return "${workConfiguration.notificationConfiguration.resourceUrl}/$resourceId"
   }
+
+  fun findMatchingAttribute(key: String, values: List<Any>): String? {
+    try {
+      val fieldValue = getProperty(key) as? String
+      if (propertyMatches(values, fieldValue)) {
+        return fieldValue
+      }
+    } catch (e: IllegalArgumentException) {
+      return null
+    }
+
+    return null
+  }
+
+  private fun propertyMatches(values: List<Any>, fieldValue: String?): Boolean {
+    if (fieldValue == null) {
+      return false
+    }
+    val splitFieldValue = fieldValue.split(",").map { it.trim() }
+
+    return values.contains(fieldValue) ||
+      values.any { it is String && fieldValue.patternMatched(it) || splitFieldValue.contains(it) }
+  }
+
+  private fun <R : Any?> getProperty(propertyName: String): R {
+    try {
+      return readPropery(propertyName)
+    } catch (e: NoSuchElementException) {
+      val details: Map<String, Any?>? = readPropery("details")
+      if (details != null) {
+        return details[propertyName] as R
+      }
+
+      throw e
+    }
+  }
+
+  private fun <R : Any?> readPropery(propertyName: String): R {
+    @Suppress("UNCHECKED_CAST")
+    return javaClass.kotlin.memberProperties.first { it.name == propertyName }.get(this) as R
+  }
+
+  fun String.patternMatched(p: String): Boolean =
```

Comment:
I think we can add a comment with a sample rule here , i think it would make easier to read